Quarkus consul quarkus.consul-config.agent.host-port configuration

I use windows 10 with Docker toolbox and I have a docker consul on port 8095. When I raise my application pointing to 192.168.99.100:8095, internally it changes to 192.168.99.100/192.168.99.100:80 as if it validated if that host existed in that port.

This is my configuration:

quarkus.consul-config.agent.host-port=192.168.99.100:8095

SO: Windows 10 home 
Java: 11.0.6.hs-adpt 
Quarkus: 1.5.0.Final

The code I have made works both in windows and in linux and even in native for linux, even before adding consul.

Conflict library: quarkus-consul-config

I have tried the same with spirng and if it works when bringing the properties.

Caused by: org.apache.http.conn.HttpHostConnectException: Connect to 192.168.99.100:80 [/192.168.99.100] failed: Connection refused: connect at org.apache.http.impl.conn.DefaultHttpClientConnectionOperator.connect(DefaultHttpClientConnectionOperator.java:156) at org.apache.http.impl.conn.PoolingHttpClientConnectionManager.connect(PoolingHttpClientConnectionManager.java:376) at org.apache.http.impl.execchain.MainClientExec.establishRoute(MainClientExec.java:393) at org.apache.http.impl.execchain.MainClientExec.execute(MainClientExec.java:236) at org.apache.http.impl.execchain.ProtocolExec.execute(ProtocolExec.java:186) at org.apache.http.impl.execchain.RetryExec.execute(RetryExec.java:89) at org.apache.http.impl.execchain.RedirectExec.execute(RedirectExec.java:110) at org.apache.http.impl.client.InternalHttpClient.doExecute(InternalHttpClient.java:185) at org.apache.http.impl.client.CloseableHttpClient.execute(CloseableHttpClient.java:83) at org.apache.http.impl.client.CloseableHttpClient.execute(CloseableHttpClient.java:108) at io.quarkus.consul.config.runtime.DefaultConsulConfigGateway.getValue(DefaultConsulConfigGateway.java:130) at io.quarkus.consul.config.runtime.ConsulConfigSourceProvider.getConfigSources(ConsulConfigSourceProvider.java:55)
![1](https://user-images.githubusercontent.com/53880516/83962603-4cfbfc80-a864-11ea-8015-17225e25587a.PNG)
![2](https://user-images.githubusercontent.com/53880516/83962604-4d949300-a864-11ea-9b18-a25e06a9a45e.png)
![3](https://user-images.githubusercontent.com/53880516/83962605-4d949300-a864-11ea-9aa9-574b2523e666.png)
![4](https://user-images.githubusercontent.com/53880516/83962606-4e2d2980-a864-11ea-8ea0-a0196cea9189.png)
![5](https://user-images.githubusercontent.com/53880516/83962607-4e2d2980-a864-11ea-9d38-bbf48aa23722.png)
![6](https://user-images.githubusercontent.com/53880516/83962609-4ec5c000-a864-11ea-9d6c-44ca7038f4ec.png)


What can I do in the settings so that the port change does not happen?

--- a/extensions/consul-config/runtime/src/main/java/io/quarkus/consul/config/runtime/DefaultConsulConfigGateway.java
+++ b/extensions/consul-config/runtime/src/main/java/io/quarkus/consul/config/runtime/DefaultConsulConfigGateway.java
@@ -118,7 +118,8 @@ class DefaultConsulConfigGateway implements ConsulConfigGateway {
             httpClientBuilder.setSSLSocketFactory(sslSocketFactory);
         }
         try (CloseableHttpClient client = httpClientBuilder.build()) {
-            String finalUri = (consulConfig.agent.useHttps ? "https" : "http") + "://" + consulConfig.agent.hostPort
+            String finalUri = (consulConfig.agent.useHttps ? "https" : "http") + "://"
+                    + consulConfig.agent.hostPort.getHostName() + ":" + consulConfig.agent.hostPort.getPort()
                     + "/v1/kv/"
                     + key;
             HttpGet request = new HttpGet(finalUri);

